x-pack/test/apm_api_integration/tests/service_overview/instances_main_statistics.spec.ts

```diff
@@ -4,436 +4,594 @@
+      endpoint: `GET /internal/apm/services/{serviceName}/service_overview_instances/main_statistics`,
+      params: {
+        path: { serviceName },
+        query: {
+          latencyAggregationType: LatencyAggregationType.avg,
+          start: new Date(start).toISOString(),
+          end: new Date(end).toISOString(),
+          transactionType: 'request',
+          environment: 'production',
+          kuery: '',
+          sortField,
+          sortDirection,
+        },
+      },
+    });
+
+    return body.currentPeriod;
+  }
 
   registry.when(
-    'Service overview instances main statistics when data is not loaded',
+    'Instances main statistics when data is not loaded',
     { config: 'basic', archives: [] },
     () => {
       describe('when data is not loaded', () => {
-        it('handles the empty state', async () => {
-          const response = await apmApiClient.readUser({
-            endpoint: `GET /internal/apm/services/{serviceName}/service_overview_instances/main_statistics`,
-            params: {
-              path: { serviceName: 'opbeans-java' },
-              query: {
-                latencyAggregationType: LatencyAggregationType.avg,
-                transactionType: 'request',
-                start: moment(end).subtract(15, 'minutes').toISOString(),
-                end,
-                offset: '15m',
-                environment: 'ENVIRONMENT_ALL',
-                kuery: '',
-              },
-            },
-          });
-          expect(response.status).to.be(200);
-          expect(response.body.currentPeriod).to.eql([]);
-          expect(response.body.previousPeriod).to.eql([]);
+        it('handles empty state', async () => {
+          const response = await getServiceOverviewInstancesMainStatistics({ serviceName: 'foo' });
+          expect(response).to.eql({});
         });
       });
     }
   );
 
   registry.when(
-    'Service overview instances main statistics when data is loaded without comparison',
-    { config: 'basic', archives: [archiveName] },
+    'Instances main statistics when data is loaded',
+    { config: 'basic', archives: [] },
     () => {
-      describe('fetching java data', () => {
-        let response: {
-          body: APIReturnType<`GET /internal/apm/services/{serviceName}/service_overview_instances/main_statistics`>;
-        };
-
-        beforeEach(async () => {
-          response = await apmApiClient.readUser({
-            endpoint: `GET /internal/apm/services/{serviceName}/service_overview_instances/main_statistics`,
-            params: {
-              path: { serviceName: 'opbeans-java' },
-              query: {
-                latencyAggregationType: LatencyAggregationType.avg,
-                start,
-                end,
-                transactionType: 'request',
-                environment: 'ENVIRONMENT_ALL',
-                kuery: '',
-              },
-            },
-          });
+      describe('Order by error rate', () => {
```

Comment:
Thanks for adding this. It was exactly what I had hoped when I saw your change.

... only thing that we might want to add, is to test the case where there are more instances than we can expose via the API. I think the size is 1000 meaning we only fetch the top 1000 instance.
We could either have the synthtrace scenario that produces 1500 instances, or we could make the size property configurable via the API and that way reduce it to something much lower (eg 50) - then the scenario just need to produce 100 instances.

Reply:
1000? Are you sure? On this PR I changed it from `50` to `100` instances (https://github.com/elastic/kibana/pull/174164/files#diff-1b2e87824a6770d2e9de6d6cfd6822dd78f0b33a36c1e68f94a35ff6b18dabc7R47). Am I missing something?

Reply:
I was just guessing - I thought it would be way higher :)